cxx/hirm_test.cc

```diff
@@ -0,0 +1,65 @@
+// Apache License, Version 2.0, refer to LICENSE.txt
+
+#define BOOST_TEST_MODULE test HIRM
+
+#include "hirm.hh"
+
+#include <boost/test/included/unit_test.hpp>
+#include <random>
+
+#include "util_distribution_variant.hh"
+namespace tt = boost::test_tools;
+
+BOOST_AUTO_TEST_CASE(test_hirm) {
+  std::map<std::string, T_relation> schema1{
+      {"R5",
+       T_noisy_relation{
+           {"D1", "D1", "D2"}, EmissionSpec("sometimes_bitflip"), "R1"}},
+      {"R1", T_clean_relation{{"D1", "D1"}, DistributionSpec("bernoulli")}},
+      {"R6",
+       T_noisy_relation{{"D1", "D1"}, EmissionSpec("sometimes_bitflip"), "R1"}},
+      {"R7",
+       T_noisy_relation{
+           {"D1", "D1", "D3"}, EmissionSpec("sometimes_bitflip"), "R6"}},
+      {"R2", T_clean_relation{{"D1", "D2"}, DistributionSpec("normal")}},
+      {"R3", T_clean_relation{{"D3", "D1"}, DistributionSpec("bigram")}},
+      {"R4",
+       T_noisy_relation{
+           {"D1", "D2", "D3"}, EmissionSpec("sometimes_gaussian"), "R2"}}};
+
+  std::mt19937 prng;
+  HIRM hirm(schema1, &prng);
+
+  hirm.transition_cluster_assignments_all(&prng);
+
+  hirm.remove_relation("R3");
```

Comment:
No -- in the current code, relations are removed only when a substitute relation has been added (e.g. maybe a new "R3" was added to one IRM in an HIRM, and the base relation pointers in the noisy relations were updated to point to the new "R3", then the old "R3" is removed from a different IRM).

Reply:
Gotcha. I might add a comment to HIRMs `remove_relation` to comment about noisy relations. 

In particular, right now this is safe, since `remove_relation` on an HIRM is never called, and only `remove_relation` on the inner IRMs are called (e.g. for the Gibbs steps) to shuffle around the relations between different views. I could see a model down the line inferring the relations / reusing this code base, and might want all noisy relations that point to a clean one to disappear.

Reply:
Thanks, that makes sense. I added a comment to `HIRM::remove_relation`.